main.c

```diff
@@ -819,9 +819,71 @@ resolve_all_solvables (Pool  *pool,
   return solv_failed;
 }
 
+static void
+add_solvable_to_pile (const char *solvable,
+                      Pool       *pool,
+                      Queue      *selection,
+                      int         flags,
+                      Queue      *solvq)
+{
+  selection_make (pool, selection, solvable, flags);
+  g_auto(Queue) q;
+  queue_init (&q);
+  selection_solvables (pool, selection, &q);
+  if (!q.count)
+    {
+      g_warning ("Nothing matches '%s'", solvable);
+      return;
+    }
+  pool_best_solvables (pool, &q, 0);
+  for (int j = 0; j < q.count; j++)
+    queue_push (solvq, q.elements[j]);
+}
+
+static void
+add_solvables_from_file_to_pile (const char  *filename,
+                                 Pool        *pool,
+                                 Queue       *selection,
+                                 int          flags,
+                                 Queue       *solvq,
+                                 GError     **error)
+{
+  GIOChannel *ch = g_io_channel_new_file (filename, "r", error);
+  if (ch == NULL)
+  {
+    g_prefix_error (error, "Unable to open file '%s': ", filename);
```

Comment:
are you sure we need this?

Reply:
Without it, the message reported is `Error: No such file or directory`--with gives the user no indication of the troublesome file name. With it, the message is reported as `Error: Unable to open file 'nosuchfilefilename': No such file or directory`.